src/humanpy/initialize.py

```diff
@@ -1,120 +1,141 @@
-            raise Exception('Failed creating viewer of type "{0:s}"'.format(
-                attach_viewer))
-
-    #Remove ROS Logging since loading Rviz might have added it
-    prpy.logger.remove_ros_logger()
-
-    return env, robot
+PACKAGE = 'humanpy'
+import logging
+import os
+
+from catkin.find_in_workspaces import find_in_workspaces
+from openravepy import Environment, IkParameterizationType, RaveCreateController, RaveCreateModule
+from openravepy.databases.inversekinematics import InverseKinematicsModel
+
+from humanhand import HumanHand
+import humanpy.action # register actions
+import humanpy.tsr # register TSR libraries
+import prpy
+from prpy.base.endeffector import EndEffector
+from prpy.base.manipulator import Manipulator
+from prpy.base.robot import Robot
+from prpy.bind import bind_subclass
+from prpy.planning.base import Sequence
+from prpy.planning.cbirrt import CBiRRTPlanner
+from prpy.planning.vectorfield import VectorFieldPlanner
+from prpy.planning.snap import SnapPlanner 
+from prpy.planning.retimer import HauserParabolicSmoother
+import numpy
+
+logger = logging.getLogger('humanpy')
+
+def initialize(attach_viewer=False, sim=True, user_id='human', env=None):
+    """Initialize the Human Robot"""
+    
+    prpy.logger.initialize_logging()
+    
+    from catkin.find_in_workspaces import find_in_workspaces
+    package_name = 'humanpy'
+    directory = 'ordata/dae'
+    objects_path = find_in_workspaces(
+        search_dirs=['share'],
+        project=package_name,
+        path=directory,
+        first_match_only=True)
+    if len(objects_path) == 0:
+        print('Can\'t find directory %s/%s' % (package_name, directory))
+        sys.exit()
+    else:
+        print objects_path # for me this is '/home/spelle/storage/ros_test_ws/src/humanpy/ordata/dae/man1.dae'
+        objects_path = objects_path[0]
+    
+    if not env:
+        env = Environment()
+
+    #Setup Manipulators
+    with env:
+        robot_file = os.path.join(objects_path, 'man1.dae')
+        robot = env.ReadKinBodyXMLFile(robot_file)
+        robot.SetName(user_id)              #needed in order to have different humans in the same env
+        env.AddKinBody(robot) 
+
+        robot.left_arm = robot.GetManipulator('leftarm')
+        robot.left_arm.hand = robot.left_arm.GetEndEffector()
+        robot.left_hand = robot.left_arm.hand
+
+        robot.right_arm = robot.GetManipulator('rightarm')
+        robot.right_arm.hand = robot.right_arm.GetEndEffector()
+        robot.right_hand = robot.right_arm.hand
+
+        bind_subclass(robot, Robot, robot_name=user_id)
+        bind_subclass(robot.left_arm, Manipulator)
+        bind_subclass(robot.right_arm, Manipulator)
+    
+        bind_subclass(robot.left_arm.hand, HumanHand, manipulator=robot.left_arm, sim=True)
+        bind_subclass(robot.right_arm.hand, HumanHand, manipulator=robot.right_arm, sim=True)
+        
+        #Setup Controller
+        with env:

```

Comment:
This is already in a `with env` block - you don't need to lock the environment again. :smile: